lib/devise_saml_authenticatable/model.rb

```diff
@@ -71,7 +71,7 @@ def find_for_shibb_authentication(conditions)
         end
 
         def attribute_map
-          @attribute_map ||= attribute_map_for_environment
+          @attribute_map ||= Devise.attribute_map || attribute_map_for_environment
```

Comment:
We could probably just use `Devise.attribute_map` without any backup, and load the value of the config file into this variable in an initializer railtie, something like

```ruby
require 'rails'
 
module DeviseSamlAuthenticatable
  module Railtie
    class AttributeMapInitializer < Rails::Railtie
      initializer "devise_saml_authenticatable.attribute_map" do |app|
        Devise.attribute_map ||= attribute_map_for_environment
      end

      def attribute_map_for_environment
        # move the implementation from this class
      end
    end
  end
end
```